src/promo/stories/messageTile.jsx

```diff
@@ -0,0 +1,72 @@
+import ComponentOverview from '../../ComponentOverview'
+import React from 'react'
+import { ThemeProvider } from '../../styles'
+import Typography from '../../Typography'
+import MessageTile, { MessageTile as UnwrappedMessageTile } from '../MessageTile'
+import { action } from '@storybook/addon-actions'
+
+import by from './by.svg'
+import cc from './cc.svg'
+import nd from './nd.svg'
+
+import accentTheme from '../../styles/themes/accent'
+import coreTheme from '../../styles/themes/core'
+import defaultTheme from '../../styles/themes/default'
+
+export default () => (
+  <ComponentOverview heading='MessageTile' component={UnwrappedMessageTile}>
+    <div style={{ width: 220 }}>
+      <ThemeProvider theme={defaultTheme()}>
+        <MessageTile
+          title={
+            <React.Fragment>
+              <img style={{ width: 26, marginRight: 6, marginBottom: 8 }} src={by} />
+              <img style={{ width: 26, marginRight: 6, marginBottom: 8 }} src={cc} />
+              <img style={{ width: 26, marginBottom: 8 }} src={nd} />
+              <div>We believe in the free flow of information.</div>
+            </React.Fragment>
+          }
+          body='All our articles can be republished for free, online or in print, under the Creative Commons licence.'
+          buttonText='Republish for free'
+          onClick={action('clicked')}
+        />
```

Comment:
The `MessageTile` usage reminds me of MUI's `Dialog`. There are components within it which have to have their specific styles and events, but they should be flexible enough to support components within them. I don't think that `MessageTile`'s inner components (`title`, `body`, `button`) should be passed as props.

What do you think about this API?

```jsx
<MessageTile>
  <MessageTileHeader>
    <img style={{ width: 26, marginRight: 6, marginBottom: 8 }} src={by} />
    <img style={{ width: 26, marginRight: 6, marginBottom: 8 }} src={cc} />
    <img style={{ width: 26, marginBottom: 8 }} src={nd} />
    <div>We believe in the free flow of information.</div>
  </MessageTileHeader>
  <MessageTileBody>
    All our articles can be republished for free, online or in print, under the Creative Commons licence.
  </MessageTileBody>
  <MessageTileButton onClick={action('clicked')}>
    Republish for free
  </MessageTileButton>
</MessageTile>
```

With this, you would have the root component (`MessageTile`) with its three other sub-components: `MessageTile/MessageTileHeader`, `MessageTile/MessageTileBody` and `MessageTile/MessageTileButton`.

This approach also avoids mixing button related props with the root component. That could cause a bit of confusion, as all inner components are optional.

What do you think?

Reply:
This sounds pretty good. I'd thinking I'd like to keep it in the `/src/promo` folder... but yeah, now I see the similarities to the `Dialog`.

Reply:
I've revised my opinion, and I agree putting it in`src/MessageTile` does make sense...
We have some components in their own top level folder, and some in subfolders.